security/cmd/chiron/main.go

```diff
@@ -0,0 +1,313 @@
+
+	// The minimum grace period for cert rotation.
+	certMinGracePeriod time.Duration
+
+	logOptions *log.Options
+	// Currently, no topic is registered for ctrlz yet
+	ctrlzOptions *ctrlz.Options
+
+	// The length of certificate rotation grace period, configured as the ratio of the certificate TTL.
+	// If certGracePeriodRatio is 0.2, and cert TTL is 24 hours, then the rotation will happen
+	// after 24*(1-0.2) hours since the cert is issued.
+	certGracePeriodRatio float32
+
+	// Whether enable the webhook controller
+	enableController bool
+
+	// Whether delete webhookconfigurations upon exit
+	deleteWebhookConfigurationsOnExit bool
+}
+
+func init() {
+	flags := rootCmd.Flags()
+
+	flags.BoolVar(&opts.enableController, "enable-controller", false, "Specifies whether enabling "+
+		"Istio Webhook Controller.")
+	flags.BoolVar(&opts.deleteWebhookConfigurationsOnExit, "delete-webhook-configurations-on-exit", false, "Specifies whether deleting "+
+		"webhookconfigurations when Istio Webhook Controller exits.")
+	flags.StringVar(&opts.certificateNamespace, "certificate-namespace", "istio-system",
+		"The namespace of the webhook certificates.")
+
+	flags.StringVar(&opts.kubeConfigFile, "kube-config", "",
+		"Specifies path to kubeconfig file. This must be specified when not running inside a Kubernetes pod.")
+
+	// Specifies the file path to k8s CA certificate.
+	// The default value is configured based on https://kubernetes.io/docs/tasks/tls/managing-tls-in-a-cluster/:
+	// the CA certificate bundle is automatically mounted into pods using the default
+	// service account at the path /var/run/secrets/kubernetes.io/serviceaccount/ca.crt.
+	flags.StringVar(&opts.k8sCaCertFile, "k8s-ca-cert-file", "/var/run/secrets/kubernetes.io/serviceaccount/ca.crt",
+		"Specifies the file path to k8s CA certificate.")
+
+	// Certificate issuance configuration.
+	flags.Float32Var(&opts.certGracePeriodRatio, "cert-grace-period-ratio",
+		cmd.DefaultWorkloadCertGracePeriodRatio, "The certificate rotation grace period, as a ratio of the "+
+			"certificate TTL.")
+	flags.DurationVar(&opts.certMinGracePeriod, "cert-min-grace-period",
+		cmd.DefaultWorkloadMinCertGracePeriod, "The minimum certificate rotation grace period.")
+
+	flags.StringVar(&opts.webhookConfigFiles, "webhook-config-files",
+		"/etc/galley-webhook-config/validatingwebhookconfiguration.yaml,/etc/sidecarinjector-webhook-config/mutatingwebhookconfiguration.yaml",
+		"The file paths of the webhookconfigurations, separated by comma. "+
+			"Only one mutatingwebhookconfiguration file and one validatingwebhookconfiguration are supported.")
+	flags.StringVar(&opts.mutatingWebhookConfigNames, "mutating-webhook-config-names", "istio-sidecar-injector",
+		"The names of the mutatingwebhookconfiguration resources in Kubernetes, separated by comma. Currently, Chiron will only manage the first one.")
+	flags.StringVar(&opts.mutatingWebhookServiceNames, "mutating-webhook-service-names", "istio-sidecar-injector",
```

Comment:
Does it make sense to reuse the service name for the config name for the same webhook?

Reply:
I think it is ok that service name and the config name are the same for a webhook, because they are two different types of resources.